py/server/deephaven/experimental/iceberg.py

```diff
@@ -243,6 +340,95 @@ def table(self, instructions: Optional[IcebergInstructions] = None, snapshot_id:
             return IcebergTable(self.j_object.table(snapshot_id, instructions))
         return IcebergTable(self.j_object.table(instructions))
 
+    def append(self,
+               table_identifier: str,
+               tables: List[Table],
+               partition_paths: Optional[List[str]] = None,
+               instructions: Optional[IcebergParquetWriteInstructions] = None):
+        # TODO Review javadoc in this file once again
+        """
+        Append the provided Deephaven table as a new partition to the existing Iceberg table in a single snapshot. This
+        will not change the schema of the existing table.
+
+        Args:
+            table_identifier (str): the identifier string for iceberg table to append to.
+            tables (List[Table]): the tables to append.
+            partition_paths (Optional[List[str]]): the partitioning path at which data would be appended, for example,
+                "year=2021/month=01". If omitted, we will try to append data to the table without partitioning.
+            instructions (Optional[IcebergParquetWriteInstructions]): the instructions for customizations while writing.
+        """
+        builder = _JIcebergAppend.builder().tableIdentifier(table_identifier)
+
+        for table in tables:
+            builder.addDhTables(table.j_table)
+
+        for partition_path in partition_paths:
+            builder.addPartitionPaths(partition_path)
+
+        if instructions is not None:
+            builder.instructions(instructions.j_object)
+
+        return self.j_object.append(builder.build())
+
+    def overwrite(self,
```

Comment:
should this be a separate method or just take an `overwrite` argument?

Reply:
I have kept it separate since append() and overwrite() APIs is more clear and also similar to how we have kept in the Java code.